src/replication/ng/dao/execution.go

```diff
@@ -0,0 +1,377 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dao
+
+import (
+	"time"
+
+	"fmt"
+	"github.com/astaxie/beego/orm"
+	"github.com/goharbor/harbor/src/common/dao"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/replication/ng/dao/models"
+)
+
+// AddExecution ...
+func AddExecution(execution *models.Execution) (int64, error) {
+	o := dao.GetOrmer()
+
+	sql := "insert into replication_execution (policy_id, status, status_text, total, failed, succeed, in_progress, stopped, trigger) " +
+		"values (?, ?, ?, ?, ?, ?, ?, ?, ?) RETURNING id"
+
+	var id int64
+	err := o.Raw(sql, execution.PolicyID, execution.Status, execution.StatusText, execution.Total, execution.Failed,
```

Comment:
Why not use `o.Insert` ?

Reply:
use db to initialize start_time and end_time. start_time will be set to now and end_time will be nil.

Reply:
in the first design execution doesn't contain the property 'status'.  and status will be generated from other properties.